lld/MachO/ObjC.cpp

```diff
@@ -1265,10 +1297,16 @@ void ObjcCategoryMerger::removeRefsToErasedIsecs() {
 void ObjcCategoryMerger::doMerge() {
   collectAndValidateCategoriesData();
 
-  for (auto &entry : categoryMap)
-    if (entry.second.size() > 1)
+  for (auto &entry : categoryMap) {
+    if (isa<Defined>(entry.first)) {
+      // Merge all categories into the base class
+      auto *baseClass = cast<Defined>(entry.first);
+      mergeCategoriesIntoBaseClass(baseClass, entry.second);
+    } else if (entry.second.size() > 1) {
```

Comment:
Also, I'm not sure if you can do `for (auto &[sym, cats]: categoryMap)`, but that would be nice.

```suggestion
  for (auto &entry : categoryMap) {
    if (auto *baseClass = dyn_cast<Defined>(entry.first)) {
      // Merge all categories into the base class
      mergeCategoriesIntoBaseClass(baseClass, entry.second);
    } else if (entry.second.size() > 1) {
```

Reply:
`for (auto &[sym, cats]: categoryMap)` works - very nice !